Dependencies/Axiom/Verify-AssertionFailed.ps1

```diff
@@ -4,24 +4,37 @@ function Verify-AssertionFailed {
         [ScriptBlock]$ScriptBlock
     )
 
-    $assertionExceptionThrown = $false
     $err = $null
+    $assertionExceptionThrown = $false
     try {
         $null = & $ScriptBlock
     }
     catch [Exception] {
         $assertionExceptionThrown = ($_.FullyQualifiedErrorId -eq 'PesterAssertionFailed')
         $err = $_
-        $err
+        Write-Output $err
+    }
+
+    $test = & (Get-Module Pester) {
+        Get-CurrentTest
     }
 
+    if (-not $assertionExceptionThrown -and $test.ErrorRecord.Count -gt 0) {
+        $pesterAssertionErrors = $test.ErrorRecord | Where-Object { $_.FullyQualifiedErrorId -eq 'PesterAssertionFailed' }
+        $assertionExceptionThrown = $null -ne $pesterAssertionErrors
+        Write-Output $test.ErrorRecord
+    }
+
+    $test.ErrorRecord.Clear()
```

Comment:
Why are you modifying Pester's own state?

Reply:
In order to verify that each assertion is producing a corresponding error record in case of failure, we need to send the error records to the pipeline within Verify-AssertionFailed. Once we do that then, in the Test results generation stage it would double the error records results since they were being received from the pipeline and where already stored in the Test context

Reply:
You can see examples of this as part of the new tests added in Functions/Assertions/Should.Tests.ps1

Reply:
@nohwnd What is your take on this?